### PR TITLE
Add namespaces (packages) to packageInfo

### LINKED ISSUE #1: Blob storage hangs for files > about 3500 kb
Guys,
        InputStream is0 = new BufferedInputStream(new FileInputStream(f));
        Date now = new Date();
        System.out.println("uploading...@"+now+" "+f);
        String storageConnectionString="DefaultEndpointsProtocol=http"+ 
                ";AccountName=mikebell" +
                ";AccountKey=<youkey>";
        CloudStorageAccount storageAccount = CloudStorageAccount.parse(storageConnectionString);

```
    CloudBlobClient bc = storageAccount.createCloudBlobClient();
    CloudBlobContainer c=bc.getContainerReference("retain");
    CloudBlockBlob blob =c.getBlockBlobReference(f.getName());
    blob.upload(is0,f.length());
    System.out.println("File uploaded... took "+(System.currentTimeMillis()-now.getTime())+" ms");
```

hangs forever for me if file exceeds 3500 MB

Attempts to diagnose:
1. Tried many files between 3500-7000KB. All fail pass the threshold which for me was about 3500-3800KB.
2. All files that fail always fail.
3. A .NET explorer gui works when uploading
4. Windows 7 x64 laptop, plenty of RAM (8 GB).
5. JVM given 1GB of RAM via xMx
6. Run from Eclipse 3.7.1 in IDE (run from application)
7. Java 1.6.0_24 JDK (oracle)
8. Memory and Buffered Streams made no dfifference.
9. Stack trace showed many threads all blocking while uploading in httpurlconnection, main coordinator thread waiting on take (pretty much as expected for a thread pool)

### LINKED ISSUE #2: Include CHANGELOG
Guys, thanks for the hard work. But an invisible release  .1.2 is now out, with no CHANGELOG).

### PR DESCRIPTION
**Pull request by JimSuplizio**
For the Spring people on this PR: I'm opportunistically fixing [sdk/spring/spring-cloud-azure-starter-appconfiguration-config/pom.xml](https://github.com/Azure/azure-sdk-for-java/pull/41639/files#diff-1756d8fcb50cdf6023cbc6115b218c65f9d6c6b4fb68d0ee25e2fada4bb2cae6) which has been [causing a build warning because of the duplicate org.apache.maven.plugins:maven-jar-plugin definition](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4081938&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=6e1f8c23-03fe-5013-b1ae-1b014bf354dd&l=19).

The namespaces (packages) are added to the packageInfo json files prior to being processed by MSDocs. The problems with this are as follows:

1. This doesn't happen until the Docs.MS Release task at release time, which is a bit late if there's a problem.
2. The process that pulls namespaces does so from the javadoc and instead of using the one that's built, it downloads the javadoc jar, effectively making a needless network call.

The solution is to add the namespaces to the packageInfo files earlier in the process. Because I'm using the staged artifacts, the place to put this is just after the artifacts have been staged. The Fetch-Namespaces-From-Javadoc function in Docs-Toc.ps1 was updated to take the location of the javadoc jar. If the location was set, it'll use that instead of trying to download. This fixes #2 above. 

Get the namespaces to update the packageInfo during the Build job, just after artifacts are copied to staging. This fixes #1. The reason for doing this here is that there were issues with people not having skipPublishDocMs set, meaning that it's expecting to produce docs, but the library was creating an empty javadoc jar with just the README so no namespaces could be determined.

Testing was done against the [java - core - ci](https://dev.azure.com/azure-sdk/public/_build/results?buildId=4081954&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=3bed581a-3f65-5269-5360-1b6e11a4b6d2&l=14), which has libraries that publish docs, and [java - template - ci](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4081938&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=e28e7f78-806d-598e-4e2c-9dfd5946e132&l=12) whose libraries do not publish docs.

This fixes the Java part of https://github.com/Azure/azure-sdk-tools/issues/7990

Note: I'm setting skipPublishDocMs and skipPublishDocGithubIo to true for com.azure.spring:spring-cloud-azure-appconfiguration-config-web because its [POM file is producing an empty javadoc jar](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/spring-cloud-azure-appconfiguration-config-web/pom.xml#L167) so we can't determine namespaces. This [failure is known](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4080143&view=logs&j=f8e040b3-c0ff-5789-0eda-99daaaa3fc1b&t=95157bd4-d7ec-5ed7-b303-68e386ec5302&l=2194) and needs to be fixed.

Q) Why use the Javadoc Jar instead of the actual library's jar?
A) The maven-javadoc-plugin definition has exclusions, things we don't want in docs. The main plugin definition, defined in [azure-client-sdk-parent](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/parents/azure-client-sdk-parent/pom.xml#L813), has common exclusions with additional exclusions able to be defined in a library's individual pom file. This means that these classes are not part of the documentation and the exclusion is done at build time for the javadoc jar file. The library's jar would still have these classes meaning we'd have to duplicate the exclusions in some manner which would be janky, at best.

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.yml [hunk 1]
```diff
@@ -197,6 +197,16 @@ jobs:
             -Artifacts ('${{ replace(convertToJson(parameters.ReleaseArtifacts), '''', '`''') }}' | ConvertFrom-Json | Where-Object -Not skipPublishPackage )
             -InformationAction Continue
 
+      - task: Powershell@2
+        inputs:
+          filePath: $(Build.SourcesDirectory)/eng/scripts/Save-Package-Namespaces-Property.ps1
+          arguments: >
+            -ArtifactStagingDirectory $(Build.ArtifactStagingDirectory)
+            -ArtifactsList ('$(ArtifactsJson)' | ConvertFrom-Json | Select-Object name, groupId, skipPublishDocMs | Where-Object -Not "skipPublishDocMs")
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.yml:205]: I would suggest doing the filtering in the script

File: eng/scripts/Save-Package-Namespaces-Property.ps1 [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+.PARAMETER ArtifactsList
+The list of artifacts to gather namespaces for, this is only done for libraries that are
+producing docs.
+-ArtifactsList ('$(ArtifactsJson)' | ConvertFrom-Json | Select-Object name, groupId, skipPublishDocMs | Where-Object -Not "skipPublishDocMs")
+#>
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $True)]
+    [string] $ArtifactStagingDirectory,
+    # -ArtifactsList ('$(ArtifactsJson)' | ConvertFrom-Json | Select-Object name, groupId, skipPublishDocMs | Where-Object -Not "skipPublishDocMs")
```
Comment (weshaggard) [inline eng/scripts/Save-Package-Namespaces-Property.ps1:30]: Lets do this filtering in the script.

**Review by weshaggard — APPROVED** [review]
I suggest move the filtering into the script.

File: eng/pipelines/templates/jobs/ci.yml (hunk 1 above)
Comment (JimSuplizio) [reply eng/pipelines/templates/jobs/ci.yml]: Fixed with latest commit.

**Review by weshaggard — APPROVED** [review]

**Review by mrm9084 — APPROVED** [review]

**Comment by JimSuplizio** [comment]
This is already past the point where all of my changes are being tested and core is having a random failure. Since my change to sdk/spring/spring-cloud-azure-appconfiguration-config/pom.xml fixes a version SNAFU affecting every pipeline I'm going to override check-enforcer and merge

**Pull request merged**